epochStart/genesis/genesisBlockCreator.go

```diff
@@ -180,9 +182,32 @@ func CreateMetaGenesisBlock(
 	header.SetTimeStamp(args.GenesisTime)
 	header.SetValidatorStatsRootHash(args.ValidatorStatsRootHash)
 
+	err = saveGenesisMetaToStorage(args)
+	if err != nil {
+		return nil, err
+	}
+
 	return header, nil
 }
 
+func saveGenesisMetaToStorage(args ArgsMetaGenesisBlockCreator) error {
+	epochStartID := core.EpochStartIdentifier(0)
+	metaHdrStorage := args.Store.GetStorer(dataRetriever.MetaBlockUnit)
+	if check.IfNil(metaHdrStorage) {
+		return epochStart.ErrNilStorage
+	}
+	marshaledData, err := args.Marshalizer.Marshal(metaHdrStorage)
```

Comment:
You are actually storing the storer here :)

Reply:
done.